### PR TITLE
Better currentCommand subsystem logging

### PR DESCRIPTION
**Pull request by drorta**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/GBSubsystem.java [hunk 1]
```diff
@@ -12,11 +11,6 @@ public GBSubsystem(String logPath) {
 		this.logPath = logPath;
 	}
 
-	private String getCurrentCommandName() {
-		Command currentCommand = getCurrentCommand();
-		return currentCommand != null ? currentCommand.getName() : "no command is currently running on the subsystem";
-	}
-
```
Comment (YoniKiriaty) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:19]: make a function that takes a command and automatically updates the current command name variable.
additionally the variable needs to be a private variable in GBSubsystem and should have this function returning it.

**Review by YoniKiriaty — DISMISSED** [review]

File: src/main/java/frc/robot/subsystems/GBSubsystem.java [hunk 2]
```diff
@@ -1,32 +1,45 @@
...
 		subsystemPeriodic();
 	}
 
 	protected void subsystemPeriodic() {}
 
+	protected Command addNameToCommand(Command command){
+		return new SequentialCommandGroup(
+				new InstantCommand(() -> currentCommandName = command.getName()),
+				command
+		);
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:35]: use command.before()

File: src/main/java/frc/robot/subsystems/GBSubsystem.java [hunk 3]
```diff
@@ -1,32 +1,45 @@
...
+	}
+	
+	//EXAMPLE
+	public Command setPower(double power){
+		return addNameToCommand(
+				new InstantCommand(() -> Logger.recordOutput("setpower")).withName("Set power to " + power)
+		);
+	}
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:43]: remove

File: src/main/java/frc/robot/subsystems/GBSubsystem.java (hunk 2 above)
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:35]: but it doesnt solve the problem if u still use command.getName(). the purpose of the projects to not use it

File: src/main/java/frc/robot/subsystems/GBSubsystem.java [hunk 4]
```diff
@@ -1,32 +1,43 @@
...
 	}
 
 	protected void subsystemPeriodic() {}
 
+	protected Command withSetCurrentCommand(Command command){
+		return new SequentialCommandGroup(
+				new InstantCommand(() -> currentCommand = command),
+				command
+		);
+	}
```
[the remark this answers is not in the document]
Comment (NoamRosenberg08) [reply src/main/java/frc/robot/subsystems/GBSubsystem.java:41]: consider renaming to 
withCommand(Command)

File: src/main/java/frc/robot/subsystems/GBSubsystem.java [hunk 5]
```diff
@@ -1,32 +1,43 @@
...
 import org.littletonrobotics.junction.Logger;
 
 public abstract class GBSubsystem extends SubsystemBase {
 
 	private final String logPath;
+	private Command currentCommand;
 
 	public GBSubsystem(String logPath) {
 		this.logPath = logPath;
+		this.currentCommand = new InstantCommand().withName("No Command");
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:16]:
```suggestion
		this.currentCommand = new InstantCommand().withName("None");
```

File: src/main/java/frc/robot/subsystems/GBSubsystem.java [hunk 6]
```diff
@@ -1,32 +1,43 @@
...
 		subsystemPeriodic();
 	}
 
 	protected void subsystemPeriodic() {}
 
+	protected Command withSetCurrentCommand(Command command){
+		return new SequentialCommandGroup(
+				new InstantCommand(() -> currentCommand = command),
+				command
+		);
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:40]: use commad.before

File: src/main/java/frc/robot/subsystems/GBSubsystem.java [hunk 7]
```diff
@@ -1,32 +1,43 @@
...
 	@Override
 	public final void periodic() {
-		Logger.recordOutput(getLogPath() + "CurrentCommand", getCurrentCommandName());
+		Logger.recordOutput(getLogPath() + "CurrentCommand", getCurrentCommand().getName());
 		subsystemPeriodic();
 	}
 
 	protected void subsystemPeriodic() {}
 
+	protected Command withSetCurrentCommand(Command command){
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:36]:
```suggestion
	protected Command applyToSubsystem(Command command){
```
```suggestion
	protected Command asSubsystemCommand(Command command){
```
some like this,

Comment (YHerm) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:36]: why protected?

Comment (YHerm) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:36]: get name as param

**Review by YHerm — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/GBSubsystem.java [hunk 8]
```diff
@@ -23,10 +26,14 @@ public String getLogPath() {
...
 	public final void periodic() {
-		Logger.recordOutput(getLogPath() + "CurrentCommand", getCurrentCommandName());
+		Logger.recordOutput(getLogPath() + "CurrentCommand", getCurrentCommand().getName());
 		subsystemPeriodic();
 	}
 
 	protected void subsystemPeriodic() {}
 
+	public Command asSubsystemCommand(Command command) {
+		return command.beforeStarting(new InstantCommand(() -> currentCommand = command));
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:36]: add require

File: src/main/java/frc/robot/subsystems/GBSubsystem.java [hunk 9]
```diff
@@ -23,10 +26,14 @@ public String getLogPath() {
...
 	public final void periodic() {
-		Logger.recordOutput(getLogPath() + "CurrentCommand", getCurrentCommandName());
+		Logger.recordOutput(getLogPath() + "CurrentCommand", getCurrentCommand().getName());
 		subsystemPeriodic();
 	}
 
 	protected void subsystemPeriodic() {}
 
+	public Command asSubsystemCommand(String commandName) {
+		return this.currentCommand.withName(commandName);
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:36]: U need to get also command

**Review by YHerm — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/GBSubsystem.java [hunk 10]
```diff
@@ -1,20 +1,21 @@
...
 import org.littletonrobotics.junction.Logger;
 
 public abstract class GBSubsystem extends SubsystemBase {
 
 	private final String logPath;
+	private Command currentCommand;
 
 	public GBSubsystem(String logPath) {
 		this.logPath = logPath;
+		this.currentCommand = Commands.none().withName("No command is currently running on the subsystem");
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/GBSubsystem.java:13]: Just write None, it's too long

**Review by YHerm — CHANGES_REQUESTED** [review]

**Review by YHerm — APPROVED** [review]
Make sure its tested!

**Review by hillelv — APPROVED** [review]

**Review by tnrz135 — APPROVED** [review]

**Review by Nitay4 — APPROVED** [review]

**Pull request merged**